src/script/user/UserRepository.ts

```diff
@@ -592,12 +592,28 @@ export class UserRepository extends TypedEventEmitter<Events> {
     );
   }
 
-  private mapUserResponse(found: APIClientUser[], failed: QualifiedId[]): User[] {
-    const failedToLoad = failed.map(
-      /* When a federated backend is unreachable, we generate placeholder users locally with some default values */
-      userId => new User(userId.id, userId.domain),
-    );
-    const mappedUsers = this.userMapper.mapUsersFromJson(found, this.userState.self().domain).concat(failedToLoad);
+  private mapUserResponse(found: APIClientUser[], failed: QualifiedId[], dbUsers?: UserRecord[]): User[] {
+    const selfUser = this.userState.self();
+
+    if (!selfUser) {
+      throw new Error('Self user is not defined');
+    }
+
+    const selfDomain = selfUser.qualifiedId.domain;
+
+    const failedToLoad = failed.map(userId => {
+      // When a federated backend is unreachable, we try to load a user from the local database.
+      const dbUserRecord = dbUsers?.find(user => matchQualifiedIds(user.qualified_id, userId));
+
+      if (dbUserRecord && selfUser) {
+        return this.userMapper.mapUserFromJson(dbUserRecord, selfDomain);
+      }
+
+      // Otherwise, we generate placeholder users locally with some default values.
+      return new User(userId.id, userId.domain);
```

Comment:
Before returning a placeholder for "unavailable user" we try looking it up in local indexedDB because there's a chance that we've already saved this user before.